### DISCUSSION TITLE
Screen scrolls to top on appending new Masonry items

### DISCUSSION THREAD

**Discussion by jcpallise** [DerYeger/yeger#197] [Q&A]
when I add items to masonry, the page go on top. I tried to get and set the scroll y position but it's not working.
Do you have idea why ?
thanks

**Comment by DerYeger** [comment]
Is your component inside a scrolling container? If yes, you need to set the `scroll-container` prop.

I'd need to see a reproduction to give more information.

**Comment by jcpallise** [comment]
I tried with scroll container and also without. same result.
I have a long array with elements like : {id: 8, filename: 'IMG_8654.jpg', url: 'http://localhost:4000/images/8/small'}
I copy my test code : 
[Gallery.zip](https://github.com/DerYeger/yeger/files/12267051/Gallery.zip)

when I scroll down, I update the array 'visibleImages' by adding items from Images array :
 this.visibleImages = [...this.visibleImages,
          ...this.images.slice(
            this.pageNumber * this.imagesPerPage,
            (this.pageNumber + 1) * this.imagesPerPage
          ),

the scroll container is "erased" then the images are displayed with the new images but the scroll bar is reset to the top.

I don't know what I'm doing wrong...

**Reply by DerYeger** [reply]
Hmm, a possible causes could be your image components having a height of 0 during their initial render (mounting phase). This makes it impossible to restore the scroll position (which this library does indeed, check the demo page https://vue-masonry-wall.yeger.eu and remove/add items). However, a functioning distribution of items across columns would argue against that.

Again, without an executable reproduction I can't really help.
